content_scripts/hints.js

```diff
@@ -107,6 +109,9 @@ var Hints = (function(mode) {
         var hints = holder.find('>div');
         hints.each(function(i) {
             var label = $(this).data('label');
+            if (alreadyOpened.indexOf(label) >= 0) {
```

Comment:
User could use `j`/`k` to scroll in Hints mode, in which case, label will be changed for some specific link. Thus using label to identify visited links is not safe, $(this).data('link') may be a better identifier.